### PR TITLE
Switch from gpm -> dep for dependency management

### PR DESCRIPTION
**Pull request by jehiah**
This switches to use [`dep`](https://github.com/golang/dep) for dependency management.

`dep` is becoming the community standard, and solves some issues `gpm` has with windows.

This is a pretty 1:1 change, the `.travis.yml` changes to the following

```
 - wget -O dep https://github.com/golang/dep/releases/download/v0.3.2/dep-linux-amd64
 - chmod +x dep
 - ./dep ensure
```

RFR @talam @hlhendy

### CODE REVIEW

**Comment by ploxiln** [comment]
Tested, works for me on macOS with go-1.8.3

I do get an odd warning:

```
[pierce@plo-pro oauth2_proxy]$ dep ensure
dep: WARNING: Unknown field in manifest: dependencies
[pierce@plo-pro oauth2_proxy]$ dep version
dep:
 version     : v0.3.2
 build date  : 2017-10-19
 git hash    : 8ddfc8a
 go version  : go1.9
 go compiler : gc
 platform    : darwin/amd64
```

and it modified `Gopkg.lock` for some reason:

```diff
diff --git a/Gopkg.lock b/Gopkg.lock
index a1a9e2b..2acf282 100644
--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -1,4 +1,5 @@
-memo = "4e1c30cedb8c432ed4c5aa8bb57fde4948f6a59af761a4a241662adfd5e7a05c"
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
+
 
 [[projects]]
   name = "github.com/18F/hmacauth"
@@ -93,3 +94,10 @@ memo = "4e1c30cedb8c432ed4c5aa8bb57fde4948f6a59af761a4a241662adfd5e7a05c"
   packages = [".","cipher","json"]
   revision = "f8f38de21b4dcd69d0413faf231983f5fd6634b1"
   version = "v2.1.3"
+
+[solve-meta]
+  analyzer-name = "dep"
+  analyzer-version = 1
+  inputs-digest = "e3067a5a40951fccd079b55c1832fe2fcc241f541eb88c22b29ba29ad102f2ed"
+  solver-name = "gps-cdcl"
+  solver-version = 1
```

**Comment by talam** [comment]
I ran into the same issues as @ploxiln but on Ubuntu 14.04 and go1.9
```
$ dep ensure
dep: WARNING: Unknown field in manifest: dependencies
```
```diff
$ git diff
diff --git a/Gopkg.lock b/Gopkg.lock
index a1a9e2b..2acf282 100644
--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -1,4 +1,5 @@
-memo = "4e1c30cedb8c432ed4c5aa8bb57fde4948f6a59af761a4a241662adfd5e7a05c"
+# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
+
 
 [[projects]]
   name = "github.com/18F/hmacauth"
@@ -93,3 +94,10 @@ memo = "4e1c30cedb8c432ed4c5aa8bb57fde4948f6a59af761a4a241662adfd5e7a05c"
   packages = [".","cipher","json"]
   revision = "f8f38de21b4dcd69d0413faf231983f5fd6634b1"
   version = "v2.1.3"
+
+[solve-meta]
+  analyzer-name = "dep"
+  analyzer-version = 1
+  inputs-digest = "e3067a5a40951fccd079b55c1832fe2fcc241f541eb88c22b29ba29ad102f2ed"
+  solver-name = "gps-cdcl"
+  solver-version = 1
```

**Comment by jehiah** [comment]
RFR @talam @ploxiln @hlhendy

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,11 @@
 language: go
 go:
-  - 1.7.5
```
Comment (ploxiln) [inline .travis.yml]: I think it's not strictly necessary to drop go-1.7 support - go-1.8 is required to build dep, not to use it. But not a big deal, I'm OK with this.

**Comment by ploxiln** [comment]
LGTM

**Review by talam — APPROVED** [review]
:+1:

**Pull request merged**